recipes/binutils/all/conanfile.py

```diff
@@ -1,70 +1,512 @@
+# Run it using `python -m unittest conanfile.py`
+
+
 class BinutilsConan(ConanFile):
     name = "binutils"
-    description = "a collection of binary tools"
-    topics = ("gnu", "compilation", "tools")
-    url = "https://github.com/conan-io/conan-center-index"
-    homepage = "https://www.gnu.org/software/binutils"
+    description = "The GNU Binutils are a collection of binary tools."
     license = "GPL-2.0-or-later"
-
+    url = "https://github.com/conan-io/conan-center-index/"
+    homepage = "https://www.gnu.org/software/binutils"
+    topics = ("binutils", "ld", "linker", "as", "assembler", "objcopy", "objdump")
     settings = "os", "arch", "compiler", "build_type"
 
-    _source_subfolder = "source_subfolder"
+    _PLACEHOLDER_TEXT = "__PLACEHOLDER__"
+
+    options = {
+        "multilib": [True, False],
+        "with_libquadmath": [True, False],
+        "target_arch": "ANY",
+        "target_os": "ANY",
+        "target_triplet": "ANY",
+        "prefix": "ANY",
+    }
+
+    default_options = {
+        "multilib": True,
+        "with_libquadmath": True,
+        "target_arch": _PLACEHOLDER_TEXT,  # Initialized in configure, checked in validate
+        "target_os": _PLACEHOLDER_TEXT,  # Initialized in configure, checked in validate
+        "target_triplet": _PLACEHOLDER_TEXT,  # Initialized in configure, checked in validate
+        "prefix": _PLACEHOLDER_TEXT,  # Initialized in configure (NOT config_options, because it depends on target_{arch,os})
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    @property
+    def _settings_target(self):
+        return getattr(self, "settings_target", None) or self.settings
+
+    def export_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    def configure(self):
+        if self.options.target_triplet == self._PLACEHOLDER_TEXT:
+            if self.options.target_arch == self._PLACEHOLDER_TEXT:
+                # If target triplet and target arch are not set, initialize it from the target settings
+                self.options.target_arch = str(self._settings_target.arch)
+            if self.options.target_os == self._PLACEHOLDER_TEXT:
+                # If target triplet and target os are not set, initialize it from the target settings
+                self.options.target_os = str(self._settings_target.os)
+            # Initialize the target_triplet from the target arch and target os
+            self.options.target_triplet = _GNUTriplet.from_archos(_ArchOs(arch=str(self.options.target_arch), os=str(self.options.target_os), extra=dict(self._settings_target.values_list))).triplet
```

Comment:
@hlewin 
`os.api_level` is passed here to `_ArchOs` (through the `extra` argument)
Actually, all properties of the target settings object are passed here.